File: src/main/java/top/fpsmaster/ui/notification/NotificationManager.java
package top.fpsmaster.ui.notification;

import org.lwjgl.opengl.GL11;
import top.fpsmaster.prism.overlay.NotificationCenter;
import top.fpsmaster.ui.kit.EdgeUi;
import top.fpsmaster.utils.render.gui.GuiScale;

/** Edge event adapter; notification behavior and rendering live in Prism. */
public final class NotificationManager {
    private static final NotificationCenter CENTER = new NotificationCenter();

    private NotificationManager() {
    }

    public static void addNotification(String title, String description, float duration) {
        addNotification(title, description, NotificationCenter.Type.INFO, duration);
    }

    public static void addNotification(String title, String description, NotificationCenter.Type type, float duration) {
        CENTER.add(title, description, type, duration);
    }

    public static void drawNotifications() {
        float[] bounds = GuiScale.getFixedBounds();
        GL11.glPushMatrix();
        GuiScale.fixScale();
        EdgeUi.beginOverlay(bounds[0], bounds[1]);
        try {
            CENTER.paint(EdgeUi.frame());
        } finally {
            EdgeUi.end();
            GL11.glPopMatrix();
        }
    }
}

File: shared/java/top/fpsmaster/ui/notification/NotificationManager.kt
package top.fpsmaster.ui.notification

import org.lwjgl.opengl.GL11
import top.fpsmaster.utils.render.Render2DUtils
import java.util.concurrent.CopyOnWriteArrayList

object NotificationManager {
    val notifications: CopyOnWriteArrayList<Notification> = CopyOnWriteArrayList()

    fun addNotification(notification: Notification) {
        notifications.add(notification)
    }

    fun addNotification(title: String, desc: String, duration: Float) {
        notifications.add(Notification(title, desc, Type.INFO, duration))
    }

    fun drawNotifications() {
        GL11.glPushMatrix()
        Render2DUtils.fixScale()
        var y = 20f

        for (notification in notifications) {
            notification.draw(0f, y)
            if (notification.anim.end == 100.0 && notification.anim.value == 100.0) {
                notifications.remove(notification)
            }
            y += 40f
        }

        GL11.glPopMatrix()
    }
}
